File: app/src/main/java/com/jsqix/dongqing/gank/fragment/RefreshFragment.java
package com.jsqix.dongqing.gank.fragment;

import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.jsqix.dongqing.gank.R;
import com.lcodecore.tkrefreshlayout.TwinklingRefreshLayout;
import com.lcodecore.tkrefreshlayout.header.progresslayout.ProgressLayout;

/**
 * Created by dongqing on 2017/1/10.
 */

public abstract class RefreshFragment extends BaseFragment {
    protected RecyclerView recyclerView;
    protected TwinklingRefreshLayout refreshLayout;

    protected String categoryType;
    protected int page = 1;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.fragment_base, container, false);
        recyclerView = (RecyclerView) view.findViewById(R.id.recyclerView);
        refreshLayout = (TwinklingRefreshLayout) view.findViewById(R.id.refresh_layout);
        refreshLayout.setFloatRefresh(true);
        ProgressLayout headerView = new ProgressLayout(mContext);
        headerView.setColorSchemeColors(getResources().getIntArray(R.array.material_colors));
        refreshLayout.setHeaderView(headerView);
        refreshLayout.startRefresh();
        refreshLayout.setOverScrollBottomShow(false);
        if (getArguments() != null) {
            categoryType = getArguments().getString("type", "all");
        }
        return super.onCreateView(inflater, container, savedInstanceState);

    }
}

File: app/src/main/java/com/dongqing/gank/kotlin/ui/fragment/RefreshFragment.kt
package com.dongqing.gank.kotlin.ui.fragment


import com.dongqing.gank.kotlin.R
import com.dongqing.gank.kotlin.ui.present.BasePresent
import com.dongqing.gank.kotlin.ui.view.IBaseView
import com.dongqing.gank.kotlin.ui.view.IFloatView
import com.lcodecore.tkrefreshlayout.RefreshListenerAdapter
import com.lcodecore.tkrefreshlayout.TwinklingRefreshLayout
import com.lcodecore.tkrefreshlayout.header.progresslayout.ProgressLayout
import kotlinx.android.synthetic.main.fragment_base.*

/**
 *
 *
 */
abstract class RefreshFragment<P : BasePresent<V>, V : IBaseView> : BaseMvpFragment<P, V>(), IFloatView {
    var page = 1
    var categoryType = "all"
    var refrsh = false
    override fun getLayoutId(): Int {
        return R.layout.fragment_base
    }


    abstract fun getData()

    override fun initView() {
        refresh_layout.setFloatRefresh(true)
        var headView = ProgressLayout(mContext)
        var colors = resources.getIntArray(R.array.material_colors)
        headView.setColorSchemeColors(*colors)
        refresh_layout.setHeaderView(headView)
        refresh_layout.startRefresh()
        refresh_layout.setOverScrollBottomShow(true)
        if (arguments != null) {
            categoryType = arguments!!.getString("type", "all")
        }
        refresh_layout.setOnRefreshListener(object : RefreshListenerAdapter() {
            override fun onRefresh(refreshLayout: TwinklingRefreshLayout?) {
                page = 1
                refrsh = true
                getData()
            }

            override fun onLoadMore(refreshLayout: TwinklingRefreshLayout?) {
                page++
                refrsh = false
                getData()
            }
        })
    }

    override fun fastMoveToTop() {
        recyclerView.smoothScrollToPosition(0)
    }
}
